### ISSUE TITLE
Factorio 2.0/Space Age Changes

### ISSUE THREAD

**Issue by deniszholob** [deniszholob/factorio-cheat-sheet#402]
Creating a placeholder ticket for the upcoming 2.0/Space Age release

* If there are any things that need updating for 2.0 we can identify to fix
* Can also have a discussion on what if anything we want to include on the site for Space Age DLC

**Comment by sorahn** [comment]
Off the top of my head:

* Power Ratios on Nauvis have changed (the day length changed
* Pumps for Nuclear are different now (water makes 10x steam)
* RCU's don't exist
* Will eventually need a section for Quality
* Some 'cheat sheet' numbers for new sciences.
* Train colors for new items

**Comment by s3bash** [comment]
How will the cheetsheet handle both base and space age? Space Age is a DLC and it is possible to keep playing the old game with old recipes. 
Maybe add a toggle for base and space age, pull the data from diff places.

**Comment by deniszholob** [comment]
> How will the cheetsheet handle both base and space age? Space Age is a DLC and it is possible to keep playing the old game with old recipes. Maybe add a toggle for base and space age, pull the data from diff places.

Yeah, thats a great question. There is a couple ways, depending on how much we will end up displaying. 
* Simplest way is just a cheat sheet section just for space age if content is low (probably not).
* Mark sheets with a space age only note. Add filters/toggle/tropdown element, this will need some routing work. 
* There is actually an implementation of this already in the code, as i was trying to add some sheets for mods, but 1) there wasnt much traction 2) it messed up routing (something to do with actual host being hithub pages https://deniszholob.github.io/factorio-cheat-sheet/ vs the url being https://factoriocheatsheet.com/ causing improper routing) so i ended up disabling that
* I am also in the process of rewriting the codebase to 1) upgrade to angular 18 (latest as of now) and getting rid of a bunch of fluff code (18 is cleaner to write), 2) replacing bootstrap with tailwind, 3) figuring out routing again. As of right now i got most of the structure updated working through each of the sheet components. Its a bit tedious tho...

**Comment by ComradeBlaze** [comment]
2.0 has allowed the player to flip chemical plants and refineries to have inputs on the other sides. Under Oil Refining, the statement "Water input for Refineries is after Oil going clockwise. Water input for Chemical Plants is the opposite" is no longer applicable.

**Comment by sorahn** [comment]
Fusion Reactor / Turbine ratio calculator :D

**Comment by ComradeBlaze** [comment]
Rockets also no longer require 100 rocket parts

**Comment by deniszholob** [comment]
> Rockets also no longer require 100 rocket parts

Is this for DLC or 2.0?
I was looking at the rocket ratios, and thought i saw 100 is still required for 2.0. Need to double check

**Comment by Xelaph** [comment]
It is now also possible to redo actions by pressing `CTRL+Y`

**Comment by Everspace** [comment]
I have only been on Vulcanis at the moment, but I would have liked (ratios when listed are my best guess, I have been delirious and not really optimizing)..

- Acid Neutralization -> Electric Turbine ratio for power (3:100)
- Acid Neutralization -> Steam Condensation (1:2)
- Metallurgic Science Pack ratio
- Offshore -> Molten Copper and Iron (1:38.4)
- Maybe suggestions on what to do with all the stone 😰
- Foundry-wire and foundry-plate Green Circuit (9 Wire:8 Iron:6 Green in assembler 3)
- Foundry-wire Red Circuit (I don't wanna)

**Comment by Everspace** [comment]
Nauvis metal processing with foundry:

- 1/32calcite per Liquid Metal Foundry.
- 15plate/s (half yellow) = 16 plate foundry + 100 fluid (25 ore/s)
- 15steel/s (half yellow) = 32 plate foundry + 300 fluid (20 ore/s)

Full belt of product in terms of Foundry/Fluid Foundry/Ore:

Copper and Iron Plate:
- Yellow: 32/8.5/13.333
- Red: 64/12.8/20
- Blue: 96/25.6/41.5
- Green: 128/34.133/40

Steel:
- Yellow: 64/25.6/40
- Red: 128/51.2/80
- Blue: 192/76.8/120
- Green: 256/102.4/160

**Comment by Everspace** [comment]
Use fluid wagons to transport liquid metal elsewhere for plate and steel and then make the plates on site. 

- 4k Plates in a wagon, 7.5k in plates in liquid metals
- 4k Steel in a wagon, 5k in steel in liquid iron

I think this also eases the logistic problem of managing the location of calcite on Nauvis or other worlds where calcite may have to come in space.

**Comment by sorahn** [comment]
> > Rockets also no longer require 100 rocket parts
> 
> Is this for DLC or 2.0? I was looking at the rocket ratios, and thought i saw 100 is still required for 2.0. Need to double check

It goes up to 100%, but each craft increments 2.

**Comment by lane-ftw** [comment]
Wondering what new colors you're going to add to the https://factoriocheatsheet.com/#train-colors section.

**Comment by deniszholob** [comment]
> Wondering what new colors you're going to add to the https://factoriocheatsheet.com/#train-colors section.

Nothing for tailored as of right now, as thats more personal to me
As far as generated ones, maybe we can add more using https://matkl.github.io/average-color/ coming from this reddit thread
https://www.reddit.com/r/factorio/comments/gvyryk/improved_my_average_icon_colours_sheet_from_the/
Im open to PRs otherwise it will be a while till i get to this

**Comment by Nelyus** [comment]
Ratio for Space Age science, so far (don’t hesitate to double check)
- space science: 3 assembler mk2
- electromagnetic science: 2,5 electromagnetic plants
- metallurgic science: 1,25 foundry
- agricultural science: 1 biochamber

If we want integral ratios, we multiply everything by 4.

We might want to have a navian+space sciences sub-section and an all sciences section.

**Comment by Everspace** [comment]
The agricultural science ratio is wonk because of spoilage, and the ratios are a little funny due to many researches requiring some but not all.

You need some form of production for the Jellynut and Yumako Mash to gain a surplus on seeds, but do be careful of an overabundance of seeds, setup logistic safeguards and active providers to prevent them from building up.

## Nutrient

There's 4 seconds of power per nutrient at 100% freshness (not real), so I very conservatively price nutrient at roughly 0.5 nutrient/second for a biochamber to account for spoilage. This means that spoilage nutrient is _quite awful_ almost needing 1 nutrient/second.

### Biochamber Yumako Mash Nutrient (should be roughly ~20 Machines)
- 1 Nutrient from Spoilage (kickstarting biochambers)
- 1 Biochamber Yumako Mash
- 3 Biochamber Nutrient from Mash

## Bioflux

### Assembling 3s Bioflux
Much easier while you have a limited number of biochambers, with direct insertion.
- 1 Jellynut Processing (If you can manage, ~20% productivity reduces the waste here)
- 2 Yumako Mash
- 1 Bioflux

### Biochamber Bioflux
- 5 Yumako Mash
- 6 Bioflux
- 2 Jelly

## Science
Have 1 extra science producer or a heating tower to handle the slight weirdness of eggs not quite matching up (Each produces roughly 0.2 of an egg but in practice it's a bit funky)

- 5 Pentapod Egg : 2 Science
- 1 Bioflux : 4 Science

### Complete Science setup (Biochambers)
- 5 Mash
- 2 Jelly
- 6 Bioflux (+5 extra bioflux leftover for other uses)
- 4 Science
- 10 Pentapod Egg
- 1 Nutrient from Bioflux for exclusively the science and pentapod egg production (You need 2 stack inserters to empty it)
- The yamako mash nutrient setup from above for the bioflux producers

**Comment by SleepiestKnave5** [comment]
not sure if this is the right place to comment this or not, but as a noob I would love to see the big miner ratios added to the mining section, as an easy reference for how many are needed to fill each belt type. It would also be cool if green belts could be added to that table as well.

**Comment by deniszholob** [comment]
> Should add solar/accumulator ratios for each planet Edit: Oh already there for Vulcanus
> 
> Would also recommend adding ratios for lava processing e.g. how many pumps to iron/copper/steel & stone output assuming full throughput. Easy to underestimate how much bandwidth you need for stone disposal if you don't have foundations yet and need to belt stone to lava

Yeah not sure what the other planet ratios are right now, i think im gunna make a solar ratio calculator based on the wiki formula
https://github.com/deniszholob/factorio-cheat-sheet/issues/415

**Comment by deniszholob** [comment]
> Ratio for Space Age science, so far (don’t hesitate to double check)
> 
> * space science: 3 assembler ~mk2~ mk3
> * electromagnetic science: ~2,5~ 4,1667 electromagnetic plants
> * metallurgic science: ~1,25~ 2,0833 foundries
> * agricultural science: ~1~ 1,667 biochamber
> 
> If we want integral ratios, we multiply everything by ~4~ 12.
> 
> We might want to have a nauvian+space sciences sub-section and an all sciences section.

I think i would rather just have all the science produced at the same rate so i would need the ratios for all sciences

**Comment by deniszholob** [comment]
> The agricultural science ratio is wonk because of spoilage, and the ratios are a little funny due to many researches requiring some but not all.
> 
> You need some form of production for the Jellynut and Yumako Mash to gain a surplus on seeds, but do be careful of an overabundance of seeds, setup logistic safeguards and active providers to prevent them from building up.
> 
> ## Nutrient
> There's 4 seconds of power per nutrient at 100% freshness (not real), so I very conservatively price nutrient at roughly 0.5 nutrient/second for a biochamber to account for spoilage. This means that spoilage nutrient is _quite awful_ almost needing 1 nutrient/second.
> 
> ### Biochamber Yumako Mash Nutrient (should be roughly ~20 Machines)
> * 1 Nutrient from Spoilage (kickstarting biochambers)
> * 1 Biochamber Yumako Mash
> * 3 Biochamber Nutrient from Mash
> 
> ## Bioflux
> ### Assembling 3s Bioflux
> Much easier while you have a limited number of biochambers, with direct insertion.
> 
> * 1 Jellynut Processing (If you can manage, ~20% productivity reduces the waste here)
> * 2 Yumako Mash
> * 1 Bioflux
> 
> ### Biochamber Bioflux
> * 5 Yumako Mash
> * 6 Bioflux
> * 2 Jelly
> 
> ## Science
> Have 1 extra science producer or a heating tower to handle the slight weirdness of eggs not quite matching up (Each produces roughly 0.2 of an egg but in practice it's a bit funky)
> 
> * 5 Pentapod Egg : 2 Science
> * 1 Bioflux : 4 Science
> 
> ### Complete Science setup (Biochambers)
> * 5 Mash
> * 2 Jelly
> * 6 Bioflux (+5 extra bioflux leftover for other uses)
> * 4 Science
> * 10 Pentapod Egg
> * 1 Nutrient from Bioflux for exclusively the science and pentapod egg production (You need 2 stack inserters to empty it)
> * The yamako mash nutrient setup from above for the bioflux producers

Havent got the time to play to get to Gleba yet, will have to see whats usefull when i get there

**Comment by deniszholob** [comment]
> not sure if this is the right place to comment this or not, but as a noob I would love to see the big miner ratios added to the mining section, as an easy reference for how many are needed to fill each belt type. It would also be cool if green belts could be added to that table as well.

Done, thanks for suggestion

**Comment by krossavik-a** [comment]
The Inserter ratios section is missing the final bonus provided from the last upgrade of stacking belts (additional +1 to capacity for non-bulk or stack inserters).

**Comment by flamaster2016** [comment]
There should be an addition to the belts/inserter throughput to accommodate for stacked belts of all types. The maximum per second per belt of a single item is now 240.

**Comment by MicMara256** [comment]
In the Vulcanus section, there needs to be a mention of using the recipe acid neutralization to produce 500degree steam to be used in steam turbines to generate power.

**Comment by daz96050** [comment]
I think there should either be a new section or an addition to the section on ratios when using the new production buildings.
For instance, instead of 6 Red>1 Green>1 Wire>1.25 plastic
Using the Electromagnetic plants for Red:
`9 Red > 1 Green > 2 Copper Wire > 3 Plastic  (Chemical Plant)`
And for blue circuits:  (Including Green for Red)
`45 Blue > 36 Red > 34 Green > 38 Copper Wire > 12 Plastic (Chemical Plant)`
